libcnb-package/src/cross_compile.rs

```diff
@@ -7,95 +8,71 @@ use which::which;
 /// look for the required tools and returns a human-readable help text if they can't be found or
 /// any other issue has been detected.
 pub fn cross_compile_assistance(target_triple: impl AsRef<str>) -> CrossCompileAssistance {
-    // Background: https://omarkhawaja.com/cross-compiling-rust-from-macos-to-linux/
-    if target_triple.as_ref() == X86_64_UNKNOWN_LINUX_MUSL && cfg!(target_os = "macos") {
-        // There is more than just one binary name here since we also support the binary name for
-        // an older version cross_compile_assistance which suggested installing a different
-        // toolchain.
-        let possible_gcc_binary_names = ["x86_64-unknown-linux-musl-gcc", "x86_64-linux-musl-gcc"];
+    let target = target_triple.as_ref();
```

Comment:
Rust allows to re-declare variables with the same name. This would allow us to do: 

```suggestion
    let target_triple = target_triple.as_ref();
```

No need to call [`AsRef::as_ref`](https://doc.rust-lang.org/std/convert/trait.AsRef.html#tymethod.as_ref) everywhere but also no loss of information calling the target triple just `target` which can be ambiguous to readers.